tests/unit/test_cube_slice.py

```diff
@@ -379,9 +380,52 @@ def test_ca_dim_ind(self):
     @patch('cr.cube.measures.scale_means.ScaleMeans.margin')
     @patch('cr.cube.measures.scale_means.ScaleMeans.__init__')
     def test_scale_means_marginal(self, mock_sm_init, mock_sm_margin):
+        """Test if slice method invokes cube method."""
         mock_sm_init.return_value = None
 
         cs = CubeSlice({}, 0)
         fake_axis = Mock()
         cs.scale_means_margin(fake_axis)
         assert mock_sm_margin.called_once_with(fake_axis)
+
+    def test_shape(self):
+        """Test shape based on 'as_array' and pruning."""
```

Comment:
This looks like a good candidate for a parameterized test, maybe something like this:

```python
def it_knows_its_shape(self, shape_fixture):  # ---or named `test_shape()` if you prefer---
    cube_slice, prune, expected_shape = shape_fixture
    shape = cube_slice.get_shape(prune=prune)
    assert shape == expected_value

@pytest.fixture(params=[
    (False, None, (3, 2)),
    (True, [[True, False], [True, False], [True, False]], (3,)),
    etc.
])
def shape_fixture(self, request, cube_):
    prune, mask_map, expected_value = request.param
    cube_.ndim = 2
    cube_.as_array.return_value = np.ma.masked_array(
        np.zeros((3, 2)), mask=np.array(mask_map) if mask_map else None
    )
    cube_slice = CubeSlice(cube_, 0)
    return cube_slice_, prune, expected_value
```

This isn't complete (or even correct) in every detail, but maybe gives you an idea. I recommend making the warning test a separate test method since its form and asserts are very different.

Using the fixture allows the test to be very clear and adding new cases is trivial and doesn't duplicate code.